File: tests/gen_ios/test_combined/expected/dd-sdk-ios/Sources/Datadog/Bridge/ComplexDataStructure.swift
/*
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2016-Present Datadog, Inc.
 */

import Foundation

/**
 A data structure to test data transfer through the bridge
 - Parameters:
     - someLong: A mandatory long property
     - someString: An optional string property
     - someMap: A mandatory map property
 */
struct ComplexDataStructure{
    let someLong: Int64
    var someString: String? = nil
    let someMap: Dictionary<String, Any?>
}

File: tests/gen_android/test_combined/expected/dd-bridge-android/src/main/kotlin/com/datadog/android/bridge/ComplexDataStructure.kt
/*
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2016-Present Datadog, Inc.
 */

package com.datadog.android.bridge

/**
 * A data structure to test data transfer through the bridge
 * @param someLong A mandatory long property
 * @param someString An optional string property
 * @param someMap A mandatory map property
 */
data class ComplexDataStructure(
    val someLong: Long,
    val someString: String? = null,
    val someMap: Map<String, Any?>
)
